assets/js/index.js

```diff
@@ -0,0 +1,262 @@
+      />
+      </div>
+  </div>
+</div>`;
+
+  dashboardContainer.append(currentWeatherCard);
+};
+
+//construct forecast cards
+const renderForecastWeatherCards = (forecastData) => {
+  constructForecastCard = (each) => {
+    return `<div class="forecast-card">
+        <h5 class="forecast-card-title">${each.date}</h5>
+        <div class="forecast-card-body">
+            <div class="forecast-card-content">
+                <p class="card-text">Max Temp: ${each.max_temperature}&deg;F</p>
+                <p class="card-text">Min Temp: ${each.min_temperature}&deg;F</p>
+                <p class="card-text">Wind: ${each.wind} MPH</p>
+                <p class="card-text">Humidity: ${each.humidity}%</p>
+            </div>
+            <div class="forecast-card-img">
+                <img
+                src="https://openweathermap.org/img/w/${each.iconCode}.png"
+            />
+            </div>
+        </div>
+    </div>`;
+  };
+
+  const forecastCards = forecastData.map(constructForecastCard).join("");
+
+  const forecastCardsContainer = ` <div class="forecast-container">
+        <h3 class="forecast-title">7-Day Forecast:</h3>
+        <div class="forecast-card-container"> ${forecastCards}
+        </div>
+    </div>`;
+
+  dashboardContainer.append(forecastCardsContainer);
+};
+
+const renderWeatherCards = (weatherData) => {
+  if (weatherData) {
+    renderCurrentWeatherCard(weatherData.current);
+
+    renderForecastWeatherCards(weatherData.forecast);
+  } else {
+    renderAlertCard();
+  }
+};
+
+const renderRecentCities = () => {
+  //get cities from LS
+  const cities = JSON.parse(localStorage.getItem("recentCities")) ?? [];
+
+  cityList.empty();
+
+  const constructAndAppendCity = (city) => {
+    //construct each li item
+    const liEl = `<li data-city=${city} class="list-group-item">${city}</li>`;
+
+    //append to the list
+    cityList.append(liEl);
+  };
+
+  const handleClick = (event) => {
+    const target = $(event.target);
+
+    // if click is from li
+    if (target.is("li")) {
+      //get city data-name
+      const cityDataName = target.data("city");
+
+      //render weather inf with cityDataName
+      renderWeatherInfo(cityDataName);
+    }
+  };
+
+  cityList.on("click", handleClick);
+
+  cities.forEach(constructAndAppendCity);
+};
+
+const renderWeatherInfo = async (cityName) => {
+  //get data from API
+  const weatherData = await getWeatherData(cityName);
+
+  //remove the last city forecast
+  dashboardContainer.empty();
+
+  renderWeatherCards(weatherData);
+};
+
+const handleSearch = async (event) => {
+  event.preventDefault();
+
+  const cityName = $("#city-input").val();
+
+  if (cityName) {
+    renderWeatherInfo(cityName);
+
+    renderRecentCities();
+  }
+};
+
+const handleReady = () => {
+  //render recent cities
+  renderRecentCities();
+
+  //get cities from LS
+  const cities = JSON.parse(localStorage.getItem("recentCities")) ?? [];
```

Comment:
You have a function which does this! `getCitiesFromLs`

Reply:
Done!!!!

Reply:
Thank you very much for all the comments they were very helpful!!!